src/macrofier/types.mjs

```diff
@@ -359,14 +394,27 @@ function getSchemaShape(schema = {}, module = {}, { templateDir = 'types', name
     return insertSchemaMacros(result, schema, module, theTitle, parent, property)
   }
   else if (schema.anyOf || schema.oneOf) {
-    // borrow anyOf logic, note that schema is a copy, so we're not breaking it.
-    if (!schema.anyOf) {
-      schema.anyOf = schema.oneOf
+    const template = getTemplate(path.join(templateDir, 'anyOfSchema' + suffix))
```

Comment:
Why do we need both an `anyOfSchema` template _and_ an `anyOf` template?

Reply:
In CPP, shape and type has to be handled differently. so two templates are required. Currently anyOf is treating as string, so shape is not required. so adding one more two handle shape part